pySNOM/images.py

```diff
@@ -0,0 +1,219 @@
+        channel = self.data[channelname]
+        return channel
+    
+    # Method to set the additional informations about the measurement from the neaspec info txt
+    def setParameters(self, infodict: dict):
+        self.parameters = infodict
+        self.setMeasurementModeFromParameters()
+    
+    # Method to set the 
+    def setMeasurementModeFromParameters(self):
+        if self.parameters == None:
+            print('Load the info file first')
+        else:
+            m = self.parameters["Scan"]
+            self.setMeasurementMode(defaults.image_mode_defs[m])
+
+    def getImageFromChannel(self, channelname: str):
+        singleimage = Image()
+
+        singleimage.filename = self.filename
+        singleimage.mode = self.mode
+        singleimage.parameters = self.parameters
+        singleimage.channel = channelname
+
+        channeldata = self.getChannelData(channelname)
+        singleimage.setImageParameters(channeldata)
+        singleimage.setChannel(channelname=channelname)
+
+        return singleimage
+
+
+# Single image from a single data channel
+class Image(Data):
+    def __init__(self, channeldata = None) -> None:
+        super().__init__()
+        # Describing channel and datatype
+        self.channel = None # Full channel name
+        self.order = None   # Order, nth
+        self.datatype = None # Amplitude, Phase, Topography - Enum DataTypes
+        # Important measurement parameters from gwyddion fields
+        self.xreal = None   # Physical image width
+        self.yreal = None   # Physical image height
+        self.xoff = None    # Center position X
+        self.yoff = None    # Center position Y
+        self.xres = None    # Pixel size in X
+        self.yres = None    # Pixel size in Y
+        # Overwrite the data definition of parent class
+        self.data = None # Actual data, this case it is only a single channel
+        self.processor = Process()
+
+        if channeldata:
+            self.setImageParameters(channeldata)
+
+    # Method to set the actual data (numpy array) - overwrite parent class method
+    def setData(self, data):
+        # Set the data
+        self.data = data
+        self.processor.setData(self.data)
+        
+    def getData(self):
+        # Set the data
+        return self.data
+    
+    def setChannel(self, channelname):
+        self.channel = channelname
+        self.order = int(self.channel[1])
+
+        if self.channel[2] == 'P':
+            self.datatype = DataTypes["Phase"]
+        elif 'Z' in self.channel:
+            self.datatype = DataTypes["Topography"]
+        else:
+            self.datatype = DataTypes["Amplitude"]
+
+    def setImageParameters(self, singlechannel):
+        # Set the basic attributes from gwyddion field
+        for key in singlechannel:
+            if key in dir(self):
+                setattr(self, key, singlechannel[key])
+        self.setData(data=singlechannel.data)
+
+class Process:
```

Comment:
Why are all possible operations in a single class? A more natural separation would be one operation per class and then a way to combine then. These classes should then follow the same interface for calling them.

Also, I think these preprocessors should not have a setData function, or anything that stores the data before you do operations. In CS, we call this is a class state, and we should minimize Here is an example why this is confusing:
```
p = Process()
p.setData(daa)
p.line_lever()
p.bg_polyfit()
```

So what is self.output? (BTW, self.outpus should not exist, it is also a state) We both know that this is the result of the bg_polyfit on original data, but given how this is called, I would actually assume both operations are done sequentially, so first line level and then, on the resulting data, polyfit.

This is avoided if there was no storing of data. An example of an interface that would be good:

```
d1 = LineLevel(props).process(data)
d2 = BgPolyfit().process(d1)
```

And, as in previous comment, I understand that sometimes you can not avoid having a state. But if you need, as for file properties in Data, make it as constant and as impossible to change after the class is initialized as possible.